src/Create.js

```diff
@@ -1,10 +1,85 @@
 import React, { Component } from 'react';
+import { getTypes, createDessert } from './fetch-utils.js';
+import classNames from 'classnames';
 
 // const URL = 'https://aqueous-coast-92629.herokuapp.com/desserts';
 
 class Create extends Component {
+    state = {
+            name: '',
+            icing: '',
+            type_id: 0,
+            message: '',
+            error: false,
+         };
+         componentDidMount = async () => {
+            const types = await getTypes();
+            this.setState({ types });
+        };
+
+        getDessertId = () => {
+            const dessertObject = this.state.desserts.find(
+                (dessert)=>dessert.name === this.state.network);
+            return dessertObject.id;
+        };
+
+        handleClick = async (e) =>{
+            e.preventDefault();
+            const dessertData = {
+                id: this.state.id,
+                name: this.state.name,
+                type: this.state.type_id
+            }
```

Comment:
Suuuuuuuper close here! Remember when you are creating a new item with the `CREATE` query and `POST` route, you won't have an ID assigned yet as the entry doesn't exist yet. You also want to make sure that the shape of the object you are sending through matches that of your data structure. 

Meaning, your object should contain a key-value pair for each of the columns you declared in your initial setup of your database (i.e. `name`, `icing`, & `type_id`). Once the request is received by your backend API, an `id` will be automatically generated by SQL for you.